homeassistant/components/caldav/todo.py

```diff
@@ -65,16 +84,7 @@ async def async_update(self) -> None:
                 include_completed=True,
             )
         )
-        todos = [
-            item.instance.vtodo for item in results if hasattr(item.instance, "vtodo")
-        ]
+        todo_items = [_todo_item(resource) for resource in results]
         self._attr_todo_items = [
-            TodoItem(
-                summary=get_attr_value(todo, "summary") or "",
-                status=TODO_STATUS_MAP.get(
-                    get_attr_value(todo, "status") or "",
-                    TodoItemStatus.NEEDS_ACTION,
-                ),
-            )
-            for todo in todos
+            todo_item for todo_item in todo_items if todo_item is not None
```

Comment:
I'd do one iteration over the results to create the final list instead of two iterations.

Reply:
Updated. I didn't realize i could do the alligator and list comprehension together.